### DISCUSSION TITLE
Mention doesn't get deleted when using ctrl + backspace

### DISCUSSION THREAD

**Discussion by ArditBaloku** [afconsult/quill-mention#322] [Q&A] [answered]
When using ctrl+backspace to delete a mention, it gets removed from the UI but the element still remains.
![image](https://user-images.githubusercontent.com/43915132/102983977-2c984480-450d-11eb-95fd-efb38d8c29c1.png)
![image](https://user-images.githubusercontent.com/43915132/102984014-3d48ba80-450d-11eb-81c0-fe2ed4e4676f.png)
Only the inner span gets removed, leaving the outer span existing which causes ghost mentions in our system

**Comment by mathieutu** [comment]
(same here, and you can see it well with styling)

<img width="378" alt="Screenshot 2021-01-21 at 06 45 28" src="https://user-images.githubusercontent.com/11351322/105285266-513b1700-5bb4-11eb-877b-8b838844fd90.png">
<img width="370" alt="Screenshot 2021-01-21 at 06 45 34" src="https://user-images.githubusercontent.com/11351322/105285268-5304da80-5bb4-11eb-88ce-0b3e43dcbac3.png">

**Comment by bdovh** [comment]
For me, it is neither deleted on Delete or fn-delete

**Comment by zehawki** [comment]
Why not just backspace to delete? Works just fine.

**Comment by mathieutu** [comment]
@zehawki ctrl+backspace (or alt+backspace on mac) is the shortcut to erase a whole word, so it's a quite used shortcut, and it feels natural to use it to erase a mention.

**Comment by Kolja-Strohm** [comment]
I have just found a workarround for this Issue.
I just had to insert this code into my Angular component:
```
        // begin of workaround for quill ctrl + Backspace bug
	private observer: MutationObserver = new MutationObserver((mutation: MutationRecord[]) => {
		console.log(mutation);
		mutation.forEach((m: MutationRecord) => {
			if (
				m.target.nodeName === 'SPAN' &&
				m.removedNodes.length === 1 &&
				m.removedNodes[0].nodeName === 'SPAN'
			) {
				const span: HTMLSpanElement = m.removedNodes[0] as HTMLSpanElement;
				if (span.innerHTML.startsWith('<span class="ql-mention-denotation-char"></span>')) {
					this.correctQuillMentions(span.innerText);
				}
			}
		});
	});
	private quillEditorComponent: QuillEditorComponent;
	@ViewChild('quillInput', { read: QuillEditorComponent }) public set quillInput(
		value: QuillEditorComponent
	) {
		if (value?.elementRef?.nativeElement) {
			this.quillEditorComponent = value;
			console.log('observing: ' + value?.elementRef?.nativeElement.toString());
			this.observer.observe(value.elementRef.nativeElement, {
				attributes: false,
				childList: true,
				characterData: true,
				subtree: true,
			});
		}
	}

	private correctQuillMentions(nameOfRemovedMention: string): void {
		const collection: HTMLCollectionOf<Element> = document.getElementsByTagName('SPAN');
		for (let i: number = 0; i < collection.length; i++) {
			if (collection[i].getAttribute('data-value') === nameOfRemovedMention) {
				if (!collection[i].innerHTML.includes(nameOfRemovedMention)) {
					collection[i].remove();
				}
			}
		}
	}
	// end of workaround
```
This will observe content changes inside of the editor and delete the outer part of the mention if the inner part gets deleted.
Depending on your mention configuration you may need to change something to make it work.

**Comment by Claypuppet** [comment, marked as answer]
I just encountered this issue too and managed to fix it by setting contenteditable=false on the node. This can be done by extending the MentionBlot.

```
const MentionBlot = Quill.import('formats/mention');

class FixedMentionBlot extends MentionBlot {
  static create(data) {
    const node = super.create(data);
    node.contentEditable = 'false';
    return node;
  }
}

Quill.register('formats/mention', FixedMentionBlot, true);
```
